cpp/src/arrow/record_batch.cc

```diff
@@ -382,4 +382,13 @@ Result<std::shared_ptr<RecordBatchReader>> RecordBatchReader::Make(
   return std::make_shared<SimpleRecordBatchReader>(std::move(batches), schema);
 }
 
+Result<std::shared_ptr<RecordBatchReader>> RecordBatchReader::Make(
+    Iterator<std::shared_ptr<RecordBatch>> it, std::shared_ptr<Schema> schema) {
+  if (schema == nullptr) {
+      return Status::Invalid("Cannot infer schema from nullptr");
```

Comment:
@rui-mo 

The term `infer` here doesn't seem to be accurate based on the code context?

Reply:
Yes, here the schema is needed. How about 'Valid schema is required' or something else?